docs/toolchains.md

```diff
@@ -467,3 +467,35 @@ Currently the following flags are used to influence toolchain selection:
 * {obj}`--@rules_python//python/config_settings:py_linux_libc` for selecting the Linux libc variant.
 * {obj}`--@rules_python//python/config_settings:py_freethreaded` for selecting
   the freethreaded experimental Python builds available from `3.13.0` onwards.
+
+## Accessing the underlying interpreter
+
+To access the interpreter that bazel manages, you can use the
+`@rules_python//python/bin:interpreter` target. This is a binary target with
+the executable pointing at the `python3` binary plus the relevent runfiles.
+
+```
+$ bazel run @rules_python//python/bin:interpreter
+Python 3.11.1 (main, Jan 16 2023, 22:41:20) [Clang 15.0.7 ] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+>>>
+$ bazel run @rules_python//python/bin:interpreter --@rules_python//python/config_settings:python_version=3.12
+Python 3.12.0 (main, Oct  3 2023, 01:27:23) [Clang 17.0.1 ] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+>>>
+```
+
+You can also access a specific binary's interpreter this way by using the
+`@rules_python//python/bin:interpreter_src` target.
+
+```
+$ bazel run @rules_python//python/bin:interpreter --@rules_python//python/bin:interpreter_src=//path/to:bin
```

Comment:
Need to find an "aha!" example for this.

Reply:
What about using the publishing `twine` binary? Something that exists within `rules_python` could do the trick.

Reply:
True, but I am looking for something that would produce different results from the default. AFAICT if I leave out `--@rules_python//python/bin:interpreter_src=//path/to:bin` then it'll default to the default Python toolchain. Since the twine would also use the default Python toolchain, it wouldn't really change the outcome.

@rickeylev , could you provide insight here? What use case are you covering with pulling the toolchain from a py_binary. I suspect I'm missing something simple here.

Reply:
The case I had in mind is a versioned py_binary. In such a case, the binary may have a different version from the top-level config.

Reply:
`dev_pip//sphinx` is available for `3.11` and `3.13.0`, we could use that.